zulipterminal/model.py

```diff
@@ -516,6 +516,59 @@ def send_stream_message(self, stream: str, topic: str, content: str) -> bool:
             notify_if_message_sent_outside_narrow(composition, self.controller)
         return message_was_sent
 
+    def is_unauthorised_to_post_in_stream(self, stream_id: int) -> Optional[str]:
+        """
+        Identify whether a user is authorised to send message to the given
+        stream or not using stream_post_policy and/or is_announcement_policy
+        and accordingly update the footer
+        """
+        stream_dict = self.stream_dict[stream_id]
```

Comment:
Okay, I'll rename it to `stream`.

I will add a case within the `is_unauthorised_to_post` function to check whether the `stream_id` is valid or not and return a string in case it is invalid.